lib/compilation_engine.rb

```diff
@@ -264,10 +267,15 @@ def compile_term
 
       elsif symbol_token?(".")
         output_token # .
+
+        subroutine_name = @tokenizer.identifier
         output_token # subroutineName
+
         output_token # (
         compile_expression_list
         output_token # )
+
+        @vm_writer.write_call("#{name}.#{subroutine_name}", @expressions_count)
```

Comment:
Still [nervous](https://github.com/wildmaples/jack-compiler/pull/17/files#r785951200) about `@expressions_count` in the general case, but this is working fine for now. 🎉

Reply:
I think we'll have to care about this in the next iteration/PR as this subroutine calls another subroutine.